agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -436,8 +443,12 @@ func (cs *CredentialSpecResource) handleSSMCredentialspecFile(originalCredential
 
 	ssmParamData := ssmParamMap[ssmParam]
 
-	localCredSpecFilePath := fmt.Sprintf("%s/ssm_%s_%s.json", cs.credentialSpecResourceLocation, cs.taskARN, ssmParam)
-
+	taskArnSplit := strings.SplitN(cs.taskARN, "/", 2)
```

Comment:
Neat! You will need to also handle the case for long task arns:

```
// your code covers this one
arn:aws:ecs:us-west-2:123456789012:task/12345-678901234-56789

// this is also possible if customers opt in
arn:aws:ecs:us-west-2:123456789012:task/MyCluster/1234567890123456789
```

At best, your code works but creates a directory called "MyCluster" which may be inconsistent with what you expect. It's also possible that it fails in the long arn case.

Reply:
How about using `filepath.Base` here?

Reply:
You could also try using the aws sdk's `arn.Parse`

Reply:
Good catch! I addressed it.